src/hooks/usePanelGeomBuilder.ts

```diff
@@ -0,0 +1,146 @@
+import { useState, useEffect, useMemo, useRef } from "react"
+import type { AnyCircuitElement, PcbBoard } from "circuit-json"
+import { su } from "@tscircuit/circuit-json-util"
+import * as THREE from "three"
+import {
+  boardMaterialColors,
+  colors as defaultColors,
+} from "../geoms/constants"
+import { manifoldMeshToThreeGeometry } from "../utils/manifold-mesh-to-three-geometry"
+import type { ManifoldToplevel } from "manifold-3d/manifold.d.ts"
+import { createPanelGeometry } from "../utils/manifold/create-panel-geometry"
+
+export interface PanelGeoms {
+  panel?: {
+    geometry: THREE.BufferGeometry
+    color: THREE.Color
+    material: PcbBoard["material"]
+  }
+}
+
+interface UsePanelGeomBuilderResult {
+  geoms: PanelGeoms | null
+  pcbThickness: number | null
+  error: string | null
+  isLoading: boolean
+  panelData: any | null
+}
+
+export const usePanelGeomBuilder = (
+  manifoldJSModule: ManifoldToplevel | null,
+  circuitJson: AnyCircuitElement[],
+): UsePanelGeomBuilderResult => {
+  const [geoms, setGeoms] = useState<PanelGeoms | null>(null)
+  const [pcbThickness, setPcbThickness] = useState<number | null>(null)
+  const [error, setError] = useState<string | null>(null)
+  const [isLoading, setIsLoading] = useState<boolean>(true)
+
+  const manifoldInstancesForCleanup = useRef<any[]>([])
+
+  const panelData = useMemo(() => {
+    const panels = circuitJson.filter((e) => e.type === "pcb_panel")
+    if (panels.length === 0) {
+      return null
+    }
+    return panels[0] as any
+  }, [circuitJson])
+
+  const boards = useMemo(() => su(circuitJson).pcb_board.list(), [circuitJson])
+
+  useEffect(() => {
+    if (!manifoldJSModule || !panelData) {
+      setGeoms(null)
+      setPcbThickness(null)
+      setIsLoading(false)
+      return
+    }
+
+    setIsLoading(true)
+    setError(null)
+    const Manifold = manifoldJSModule.Manifold
+    const CrossSection = manifoldJSModule.CrossSection
+
+    const safeDelete = (inst: any) => {
+      if (!inst || typeof inst.delete !== "function") return
+      try {
+        inst.delete()
```

Comment:
Instead of copying this logic, you should create a hook that makes the lgogic for managing manifold shared